flash/text/classification/tokenizers/base.py

```diff
@@ -0,0 +1,28 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Generator, List, Mapping, Sequence
+
+
+class BaseTokenizer:
+    def __init__(self, backbone: str, pretrained: bool):
+        self.backbone = backbone
+        self.pretrained = pretrained
+        self._is_fit = pretrained
+
+    def fit(self, hf_dataset: Sequence[Mapping[str, Any]], input: str) -> None:
+        pass
+
+    def _batch_iterator(self, dataset: Sequence[Mapping[str, Any]], input: str) -> Generator[List[str], None, None]:
```

Comment:
Should we use a DataLoader ?

Reply:
The creation of a batch_iterator from an hf_dataset is taken verbatim from the [HF website](https://huggingface.co/docs/tokenizers/python/latest/tutorials/python/training_from_memory.html).

Reply:
Ok,